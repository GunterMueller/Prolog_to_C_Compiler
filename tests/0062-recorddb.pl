main :-	test, fail.
main :-
	garbage_collect,
	insert_list,
	delete_list,
	garbage_collect, name(foo, _),
	delete_list.

test :-
	recorda(foo, 123, X),
	S = abc,
	recorded(foo, N, X), N == 123,
	recordz(foo, S),
	recorded(foo, 123),
	erase(X),
	recorded(foo, S2, Z),
	display(S2), nl,
	S2 == S,
	erase(Z),
	\+recorded(foo, 123),
	recorda(foo(1), [list,1]),
	recorda(foo(2), some(term), Y),
	recorda(foo(3), another),
	recorda(foo(3), another(A,B,A,B)),
	show('once:', foo(_)),
	erase(Y),
	show('twice:', foo(_)).

show(X, Y) :-
	display(X), nl,
	recorded(Y, Z), display(Z), nl,
	fail.
show(_, _).

insert_list :-
	member(X,[1,2,3,4,5,6,7,8,9,10]),
	recordz(numbers, X),
	fail.
insert_list.

delete_list :-
	recorded(numbers, X, R), display(X), nl, erase(R),
	fail.
delete_list.
