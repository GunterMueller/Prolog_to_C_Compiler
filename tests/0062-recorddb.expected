abc
once:
another(_47, _48, _47, _48)
another
some(term)
[list,1]
twice:
another(_63, _64, _63, _64)
another
[list,1]
1
2
3
4
5
6
7
8
9
10
